### PR TITLE
[#11319] Logs Interface: Support more filters + better exception tracing

### LINKED ISSUE #11319: Logs Interface: Support more filters + better exception tracing
New filters to be added:
- Filter for latency
- Filter for HTTP status
- Arbitrary filters

Exception tracing improvement:
- Source location should indicate where the exception is thrown, not where it is caught/logged at
- Should include stack trace
- Should persist the exception class name properly via JSON payload instead of wild-guessing via text payload

### PR DESCRIPTION
**Pull request by wkurniawan07**
Fixes #11319 

- Added new filters:
  Screenshot 2021-08-01 at 2 41 40 PM
  (also reduced the vertical margin of the filters to make it less stretched)
- Fixed maintainer role not getting inserted when masquerading
- Overhauled the method of exception logging in the system:
  - Added method within logger to take in a `Throwable` instance on top of a message. When done this way, the logger will instead log the source location of that `Throwable` instead of the logger's, and will persist necessary information such as the exception class and stack traces within JSON payload, under a new event `EXCEPTION_LOG`.
  - When displaying those stack traces, the messages will be hidden for non-admins.
  - `TeammatesException.toStringWithStackTrace` is the old way of inserting stack trace into the log message, and is no longer necessary. It is thus removed, along with the `TeammatesException` class.

Display for admin:
Screenshot 2021-08-01 at 2 42 03 PM

Same exception displayed for non-admin maintainer:
Screenshot 2021-08-01 at 2 42 31 PM

### CODE REVIEW

File: src/main/java/teammates/common/util/Logger.java [hunk 1]
```diff
@@ -96,13 +100,94 @@ public void warning(String message) {
...
+    }
+
+    private List<String> getStackTraceToDisplay(Throwable t) {
+        List<String> stackTraceToDisplay = new ArrayList<>();
+        for (StackTraceElement ste : t.getStackTrace()) {
+            String stClass = ste.getClassName();
+            if (stClass.startsWith("org.eclipse.jetty.servlet")) {
+                // Everything past this line is the internal workings of Jetty
+                // and does not provide anything useful for debugging
+                stackTraceToDisplay.add("...");
```
[the remark this answers is not in the document]
Comment (wkurniawan07) [reply src/main/java/teammates/common/util/Logger.java:182]: Not really. Those only say that the chain of event goes from some filter to another (`RequestTraceFilter` to `OriginCheckFilter` to `ObjectifyFilter`) before finally hitting the servlet, nothing more.

Things could go wrong at the filters of course, but if it happens, the filter will be the one that's logging the message.

**Review by ypinhsuan — COMMENTED** [review]
The changes look good. Perhaps the error logs in `logsForLocalDev.json` for local testing can be updated as well?

File: src/main/java/teammates/common/util/Logger.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (hhdqirui) [reply src/main/java/teammates/common/util/Logger.java:182]: Ok, understood.

**Review by hhdqirui — COMMENTED** [review]
The changes look good.

**Comment by wkurniawan07** [comment]
@ypinhsuan I'll do that before merging

**Pull request merged**